datalad/distribution/publish.py

```diff
@@ -343,11 +348,25 @@ def __call__(
                         'Cannot determine target sibling for %s' % (ds,))
             elif to not in ds.repo.get_remotes():
                 # unknown given remote
+                # TODO: not quite "failing" really so interfers with "inherit_settings"
                 if skip_failing:
                     lgr.warning(
                         "Unknown target sibling '%s', skipping %s",
                         to, ds)
                     ds_remote_info[ds_path] = None
+                elif inherit_settings:
+                    superds = ds.get_superdataset()
+                    if not superds:
+                        raise RuntimeError(
+                            "%s has no super-dataset to inherit settings for the remote %s"
+                            % (ds, to)
+                        )
+                    # XXX due to difference between create-sibling and create-sibling-github
+                    # would not be as transparent to inherit for -github
+                    lgr.info("Will try to create a sibling inheriting settings from %s", superds)
+                    # XXX explicit None as sshurl for now
+                    ds.create_sibling(None, name=to, inherit_settings=True)
```

Comment:
overall idea here was to improve UX -- I got tired of remembering either I have new sub-datasets or not before running `publish`, and then invoking `create-sibling` manually in case if there were and publish failed...  I had to remember parameters for `create-sibling` invocation etc -- just doesn't scale.  Hence I have decided to address at least this common use-case where I have a hierarchy of datasets, which I could publish over and over again efficiently even if I get new sub-datasets without remembering again and again how to create those remote again and again.
Such scheme indeed might fail in case if there is two sets of hierarchies which try to sync/publish  to the common one -- that is where `existing=skip` would be necessary I guess.  For now `existing`  here indeed remained to its safest default and I did not proxied any more settings into `create_sibling` -- for any of those scenarios better to use it directly/consciously.
Having said all that I am yet to see this piece "in action" since yesterday had to revert to separate `create-sibling` invocation anyways and `publish` is hard to tame for me (see #1329) ATM